src/kudu/consensus/consensus_queue.cc

```diff
@@ -170,6 +171,7 @@ PeerMessageQueue::PeerMessageQueue(const scoped_refptr<MetricEntity>& metric_ent
       tablet_id_(std::move(tablet_id)),
       successor_watch_in_progress_(false),
       log_cache_(metric_entity, std::move(log), local_peer_pb_.permanent_uuid(), tablet_id_),
+      binlog_manager_(std::make_shared<Binlog>(&log_cache_)),
```

Comment:
the kudu log should be under the log_cache_ because it is a write through cache.

Current kudu log has 3 things ( the persistent file/segment organization), log reading and log index.
The Binlog implementation class, which you have used as an example would likely live in fbcode.
The current protobuf format logs will also have to implement the interface.